### PR TITLE
Added tutorial notebook

### PR DESCRIPTION
**Pull request by dhuppenkothen**
Added the tutorial notebook. I decided to leave in the plots so the notebook can be rendered online.

---
This change is [Reviewable](https://reviewable.io/reviews/dhuppenkothen/entrofy/30)

### CODE REVIEW

**Comment by bmcfee** [comment]
There's no easy way to do line comments on a notebook PR, so i'm just going to copypaste here:

> The goal was of the software's original use was rooted 

Grammar is weird here.

> First we decided on a set of applicants we wanted to have at the workshop.

Would rephrase as "would accept at the workshop" .. current wording sounds like pre-selects.

> unless that are the only categories included in your selection procedure

unless "those" are the only...

> Drop-down menues

"menus"

> Contact us!

Include a link to the project page.



---

- Cell 2: point out that `NaN` in the dataframe indicates non-response.

> These visualizations differ depending on whether your

line seems to be cut off here?

> Note: the option swarm will only work with the newest version of seaborn!

Use an explicit version number here.

> compare different columns at a glance

I'd say "attributes" here.

> By default, all categories will have the same target fractions.

Do you mean values here?

- Cell 19: This seems like brittle code, relying on sorting to connect values to keys.  Why not explicitly define a new targets dictionary and update the mapper?

> Make sure the keywords in the dictionary match the names of the columns in your file

Maybe include an example of retrieving the names from the dataframe?

> breaking ties between people with very close scores.

"Score" has not been defined.  Can we rephrase this?  Also, an example of a reasonable quantile value would be helpful here.

> we run the algorithm several times and select the best score out of n_trials runs

select the "highest scoring solution"

> For plotting a single column, use plot_fractions:

Can we have a legend entry on the dashed line for target fractions?

- Cell 49: why is this outputing a randomstate object?

**Comment by dhuppenkothen** [comment]
Updated tutorial notebook works on my computer without throwing an error. Do you have a cell number/traceback @bmcfee ?

**Comment by bmcfee** [comment]
It was the kde plots failing because a covariance matrix was singular.  Smells like a seaborn bug to me.

**Comment by dhuppenkothen** [comment]
Do you know which version of seaborn?

**Comment by bmcfee** [comment]
seaborn 0.7.0 or 0.7.1 both fail:

```python
LinAlgError                               Traceback (most recent call last)
<ipython-input-11-00bdbb9c2e31> in <module>()
      3 ax1 = entrofy.plotting.plot_correlation(df, "age", "age", ax=ax1,
      4                                         xtype="continuous",
----> 5                                         ytype="continuous", cont_type="kde")
      6 ax2 = entrofy.plotting.plot_correlation(df, "age", "age", ax=ax2,
      7                                         xtype="continuous",
...
/home/bmcfee/.local/lib/python2.7/site-packages/scipy/linalg/basic.pyc in inv(a, overwrite_a, check_finite)
    679         inv_a, info = getri(lu, piv, lwork=lwork, overwrite_lu=1)
    680     if info > 0:
--> 681         raise LinAlgError("singular matrix")
    682     if info < 0:
    683         raise ValueError('illegal value in %d-th argument of internal '

LinAlgError: singular matrix
```

I think it's certainly a seaborn issue.  This happens on both python 2.7 and 3.5.  I'm using pandas 0.18.1 also.
